sqs-tester/src/main/java/uk/gov/justice/laa/crime/microservice/sqstester/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.justice.laa.crime.microservice.sqstester;
+
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.util.WebUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
```

Comment:
You should probably extend the class `ResponseEntityExceptionHandler` such that it provides reuse of various additional out-of-the-box functionality.

org.springframework.web.reactive.result.method.annotation.ResponseEntityExceptionHandler;